prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+from prereise.gather.demanddata.transportation_electrification import const
+from prereise.gather.demanddata.transportation_electrification import mileage
+from prereise.gather.demanddata.transportation_electrification import charging
+import numpy as np
+
+
+def immediate_charging(
+    census_region, model_year, veh_range, kwhmi, power, location_strategy, veh_type
+):
+    """Immediate charging function
+
+    :param int census_region: any of the 9 census regions defined by US census bureau.
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040, 2050.
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int kwhmi: fuel efficiency, should vary based on vehicle type and model_year.
+    :param int power: charger power, EVSE kW.
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5; 
+        1-home only, 2-home and work related, 3-anywhere if possibile, 
+        4-home and school only, 5-home and work and school.
+    :param str veh_type: determine which category (LDV or LDT) to produce charging profiles for
+    :return: (*numpy.ndarray*) -- charging profiles.
+    """
+
+    # load NHTS data from function
+    if veh_type.lower() == "ldv":
+        newdata = mileage.remove_ldt(mileage.load_data(census_region))
+
+    elif veh_type.lower() == "ldt":
+        newdata = mileage.remove_ldv(mileage.load_data(census_region))
+
+    # updates the weekend and weekday values in the nhts data
+    newdata = mileage.update_if_weekend(newdata)
+
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata = newdata.reindex(
+        list(newdata.columns)
+        + [
+            "trip start battery charge",
+            "trip end battery charge",
+            "charging power",
+            "charging time",
+            "charging consumption",
+            "BEV could be used",
+            "trip number",
+        ],
+        axis="columns",
+        fill_value=0,
+    )
+
+    input_day = mileage.get_input_day(mileage.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = mileage.get_data_day(newdata)
+
+    trip_strategy = 1
```

Comment:
Do we want the user to be able to change this? If so, it might make more sense as an input with a default value, rather than hard-coded within the function.

Reply:
I changed `trip_strategy` to a default argument as you suggested since we're focusing on charging after any trip. But in the future, if someone may want to use the last-trip capability, then they'll be able to do so.